### ISSUE TITLE
error when using with babel runtime

### ISSUE THREAD

**Issue by zkochan** [SBoudrias/run-async#16]
[Babel Runtime transform](http://babeljs.io/docs/plugins/transform-runtime/) adds its own promise polyfill, when transpiling code. For some reason it doesn't work with `run-async`. Seems like `run-async` doesn't recognize the babel's promise polyfill and thinks that it is just a value of a sync function...

Maybe `is-promise` has to be update?

My solution was to use babel for transpiling the `run-async` package and allowing it to add its own promise polyfill.

**Comment by jamestalmage** [comment]
There is no reason this shouldn't work with `transform-runtime`, I think something else is going on. Can you provide some sample code, demonstrating what doesn't work?

> Babel Runtime transform adds its own promise polyfill,

That is not exactly true. `transform-runtime` actually avoids polluting global scope the way `babel-polyfill` does. It transforms this:

```js
var promise = new Promise(...);
```

to this:

```js
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _promise = require("babel-runtime/core-js/promise");

var _promise2 = _interopRequireDefault(_promise);

var promise = new _promise2.default();
```

While that is a bit convoluted, what it does is allow you to write your code as if `global.Promise` simply exists, and rewrites your code to use `core-js/promise` instead.

`babel-polyfill` works basically the same way, but instead it's basically the equivalent of:

```
global.Promise = require('babel-runtime/core-js/promise');
```

They both solve the same problem, but one pollutes global scope, the other doesn't.

**Comment by alexlyte** [comment]
I ran into a similar issue when trying to compile 'Strongloop', which utilizes this module. Idk if it's an environment issue (I'm running node v0.10.25), but when I tried to compile the project, I got an 'err Promise is not defined', which showed the line: 
 var promise = new Promise(function (resolve, reject) {
on line 24 of index.js. 
Not sure if it's a real issue or not... I cheated and just wrote in 'var Promise = require('bluebird')', and that seemed to alleviate my issue. But that's both a crappy fix and probably not doable by all users. 

FWIW
